src/Endpoint/StatusEndpoint.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\CdtClient\Endpoint;
+
+use OpenEuropa\CdtClient\Contract\TokenAwareInterface;
+use OpenEuropa\CdtClient\Model\Response\Translation;
+use OpenEuropa\CdtClient\Traits\TokenAwareTrait;
+use OpenEuropa\CdtClient\Traits\ValidationAwareTrait;
+
+/**
+ * Class StatusEndpoint
+ *
+ * Defines how the client should handle requests to the "requests/:requestyear/:requestnumber" space of the API.
+ * Implements the TokenAwareInterface to handle authentication tokens for secure communication.
+ *
+ * @see EndpointBase
+ * @see TokenAwareInterface
+ */
+class StatusEndpoint extends EndpointBase implements TokenAwareInterface
+{
+    use TokenAwareTrait;
+    use ValidationAwareTrait;
+
+    protected string $permanentId;
+
+    public function getPermanentId(): string
+    {
+        return $this->permanentId;
+    }
+
+    public function setPermanentId(string $permanentId): self
+    {
+        if (!preg_match('/^\d{4}\/[^\/]+$/', $permanentId)) {
+            throw new \InvalidArgumentException('Invalid permanent ID format (it should be formatted like 2024/1234567).');
+        }
+
+        $this->permanentId = $permanentId;
+        return $this;
+    }
+
+    public function execute(): Translation
+    {
+        [$year, $id] = explode('/', $this->permanentId);
```

Comment:
I know we could possibly use the endpoint like `/v2/requests/:permanentId` instead of `/v2/requests/:requestyear/:requestnumber` (because the id format is `2024/00001`), but I was afraid that changing the URL structure will be confusing for future users.